### DISCUSSION TITLE
streaming split file

### DISCUSSION THREAD

**Discussion by Aikanairo** [EverythingSuckz/TG-FileStreamBot#119] [Ideas]
Hi, I would like some advice on developing a function.

I often upload files larger than 4 gigabytes, these are video files.

the files are usually called that

split.mp4.001
split.mp4.002

or

split.mkv.001
split.mkv.002

if I play the first file with vlc and the other parts are in the same folder, vlc plays the video file without any problem

I would like to be able to stream a file divided into multiple parts

in each file uploaded to telegram in caption I wrote the name of the file and the part

How could I develop a similar function?

**Comment by EverythingSuckz** [comment]
> Hi, I would like some advice on developing a function.
> 
> I often upload files larger than 4 gigabytes, these are video files.
> 
> the files are usually called that
> 
> split.mp4.001
> split.mp4.002
> 
> or
> 
> split.mkv.001
> split.mkv.002
> 
> if I play the first file with vlc and the other parts are in the same folder, vlc plays the video file without any problem
> 
> I would like to be able to stream a file divided into multiple parts
> 
> in each file uploaded to telegram in caption I wrote the name of the file and the part
> 
> How could I develop a similar function?

Hi @Aikanairo , thank you for your suggestion. Yeah it might be possible but things might go the opposite way too since we will be hard coding many things (like checking if the file name or caption contains numbers at the end) and it'll need some kind of database eventually to store all the parts of a file.
